### ISSUE TITLE
React router with redux not working after updrage to 5.0.0

### ISSUE THREAD

**Issue by hamzaboularbah** [ReactTraining/history#803]
Hello, 
Im trying to setup a react redux boilerplate that use the react router with redux and rely on the **createBrowserHistory** api from the history package. everything was working fine, until i made the upgrade to 5.0.0 for it then it stopped working throwing the following error in the console when i click on a link 
`Uncaught Could not find router reducer in state tree, it must be mounted under "router"
`
PS : i'm already mounting the router reducer in my root reducer.
https://www.loom.com/share/d956b05939af4fada2ec891ae56c8853

**Comment by gerardo15** [comment]
Im also having this problem!

**Comment by emish89** [comment]
yes, me too with `connected-react-router` and redux after the upgrade from history 4 to 5

**Comment by Sen9a** [comment]
Hello guys.
After the update, my component doesn't render JSX when I am redirecting to it and doesn't throw any errors.
Thank you in advance.

**Comment by ApacheEx** [comment]
the same here - I see white-screen without any errors when call 
```js
history.replace({ search: '?whatever' });
```

Issue appears after upgrading to v5

**Comment by readikus** [comment]
Documentation for migrating from 4 -> 5 would be helpful.

**Comment by saadbinsaeed** [comment]
`history.push()` stopped working after upgrading from v4 to v5

**Comment by readikus** [comment]
I'm still working through the issues with this, but the thing that jumps out is that the location objects being passed around take on different forms:

For example: 

```
{
  hash: "",
  key: "6amvxt1u",
  pathname: "/foo",
  search: "",
  state: null
}
```
Then,
```
{
  action: "REPLACE"
  location: {
    hash: "",
    key: "hh38yy0c",
    pathname: "/bar",
    search: "",
    state: null
  }
}
```

**Comment by stage88** [comment]
I found i am getting a `Could not find router reducer in state tree, it must be mounted under "router"`, i traced it down to the missing `action` prop on the persisted route object, as @readikus mentioned above.

in selectors.js, there is code looking for it when validating if it is a router:
```
var isRouter = function isRouter(value) {
  return value != null && _typeof(value) === 'object' && getIn(value, ['location']) && getIn(value, ['action']);
};
```

**Comment by avindra** [comment]
You probably have to update to the react router v6 alpha as well, as the v5 is explicity specifying v4 history in its dependencies. More likely than not, you will have two (conflicting) history deps if you are still using react-router 5.

**Comment by joshxyzhimself** [comment]
Almost panicked but glad this can be resolved with a temporary downgrade haha.

**Comment by jarretmoses** [comment]
@avindra I did actually try upgrading the router to v6 before checking here but I am still having the same issues as marked here.

**Comment by vinniimiranda** [comment]
Same issue here

**Comment by colehart** [comment]
Same issue. Like @readikus, I'm noticing the object changing from place to place. On load, the object fulfills v4 / expected redux action / location, but on location change, the location object is being populated with {action: <action-type>, location: <string>}. Seems like a nesting problem in a v5 functions' return maybe?

**Comment by DLesage25** [comment]
Same issue here. Downgrading to 4.10.1 fixes the issue.

**Comment by tianleiyiji123** [comment]
Documentation for version 5 can be found in the docs directory. This is the current stable release. Version 5 is used in React Router version 6.

Documentation for version 4 can be found on the v4 branch. Version 4 is used in React Router versions 4 and 5.

**Comment by joshxyzhimself** [comment]
feels weird that history v5 is at @latest while react-router-dom v6 is at @next
